Tests/Unit/ZoomApiServiceTest.php

```diff
@@ -0,0 +1,265 @@
+
+    /** @test */
+    public function upcomingMeetingsReturnsValidCachedData()
+    {
+        $client = new Client();
+
+        $this->cacheMock = $this->getMockBuilder(VariableFrontend::class)->disableOriginalConstructor()->getMock();
+        $this->cacheMock->expects($this->once())
+            ->method('get')
+            ->willReturn(['cached123']);
+
+        $service = $this->getService($client);
+        $service->initializeObject();
+
+
+        $result = $service->getUpcomingMeetings();
+
+        $this->assertEquals(['cached123'], $result);
+    }
+
+    /** @test */
+    public function upcomingMeetingsWithEmptyCacheReturnsNewData()
+    {
+        $handlerStack = HandlerStack::create(
+            new MockHandler([
+                new Response(200, [], json_encode(['meetings' => ['meeting1'], 'next_page_token' => '1234567890'])),
+                new Response(200, [], json_encode(['meetings' => ['meeting2'], 'next_page_token' => ''])),
+            ])
+        );
+        $client = new Client(['handler' => $handlerStack]);
+
+        $service = $this->getService($client);
+        $service->initializeObject();
+        $this->cacheMock->method('get')->willReturn(false);
+
+
+        $result = $service->getUpcomingMeetings(true);
+
+
+        $this->assertEquals(['meeting1', 'meeting2'], $result);
+    }
+
+    /** @test */
+    public function upcomingMeetingsWithCacheExceptionReturnsNewData()
+    {
+        $handlerStack = HandlerStack::create(
+            new MockHandler([
+                new Response(200, [], json_encode(['meetings' => ['meeting1'], 'next_page_token' => '1234567890'])),
+                new Response(200, [], json_encode(['meetings' => ['meeting2'], 'next_page_token' => ''])),
+            ])
+        );
+        $client = new Client(['handler' => $handlerStack]);
+
+        $service = $this->getService($client);
+        $service->initializeObject();
+        $this->cacheMock->method('get')->willReturn(false);
+        $this->cacheMock->method('set')->willThrowException(new Exception());
+        $this->systemLoggerMock->shouldReceive('error')->once();
+
+
+        $result = $service->getUpcomingMeetings(true);
+
+
+        $this->assertEquals(['meeting1', 'meeting2'], $result);
+    }
+
+    /** @test */
+    public function getUpcomingMeetingsWithEmptyResponseThrowsException()
+    {
+        $this->expectException(ZoomApiException::class);
+
+        $handlerStack = HandlerStack::create(
+            new MockHandler([
+                new Response(200, [], ''),
+            ])
+        );
+        $client = new Client(['handler' => $handlerStack]);
+
+        $service = $this->getService($client);
+        $service->initializeObject();
+        $this->cacheMock->method('get')->willReturn(false);
+
+
+        $service->getUpcomingMeetings(true);
+    }
+
+    /**
+     * @param  \GuzzleHttp\Client  $client
+     * @return MockObject
+     */
+    private function getService(Client $client): MockObject
```

Comment:
If you change the return type of the response to `ZoomApiService|MockObject`  here (and in the DocBlock), the IDE knows what's going on here and can help with refactoring/autocomplete/....